dspace-spring-rest/src/main/webapp/js/hal/http/client.js

```diff
@@ -7,13 +7,55 @@
  */
 HAL.Http.Client = function(opts) {
     this.vent = opts.vent;
-    this.defaultHeaders = { 'Accept': 'application/hal+json, application/json, */*; q=0.01' };
-    cookie = document.cookie.match('(^|;)\\s*' + 'MyHalBrowserToken' + '\\s*=\\s*([^;]+)');
-    cookie ? this.defaultHeaders.Authorization = 'Bearer ' + cookie.pop() : '';
+    this.defaultHeaders = {'Accept': 'application/hal+json, application/json, */*; q=0.01'};
+    var authorizationHeader = getAuthorizationHeader();
+    authorizationHeader ? this.defaultHeaders.Authorization = authorizationHeader : '';
     console.log(this.defaultHeaders);
     this.headers = this.defaultHeaders;
 };
 
+function getAuthorizationHeader() {
+    var cookie = document.cookie.match('(^|;)\\s*' + 'MyHalBrowserToken' + '\\s*=\\s*([^;]+)');
+    if(cookie != undefined) {
+        return 'Bearer ' + cookie.pop();
+    } else {
+        return undefined;
+    }
+}
+function downloadFile(url) {
+    var request = new XMLHttpRequest();
+    request.open('GET', url, true);
+    request.responseType = 'blob';
+    var authorizationHeader = getAuthorizationHeader();
+    if (authorizationHeader != undefined) {
+        request.setRequestHeader("Authorization", authorizationHeader);
+    }
+    request.onload = function () {
+        // Only handle status code 200
+        if (request.status === 200) {
+            // Try to find out the filename from the content disposition `filename` value
+            var disposition = request.getResponseHeader('content-disposition');
+            var matches = /"([^"]*)"/.exec(disposition);
+            var filename = (matches != null && matches[1] ? matches[1] : 'file.pdf');
```

Comment:
One minor comment (via code review).  Is this defaulting the downloaded file to be named "file.pdf"?  If so, I think we should change that behavior, as we should never assume a specific file format (PDF).  If we must have a default name, then I think we should remove the extension here, and simply call it "file"

Reply:
This was indeed the case. I've changed the default name to 'content'